packages/react/src/auto/shadcn/inputs/ShadcnAutoInput.tsx

```diff
@@ -15,40 +16,44 @@ export const makeShadcnAutoInput = ({
   Checkbox,
   Button,
 }: Pick<ShadcnElements, "Input" | "Label" | "Checkbox" | "Button">) => {
-  function ShadcnAutoInput(props: { field: string; label?: string }) {
+  // Memoize the input makers
+  const memoizedInputs = React.useMemo(
+    () => ({
+      idInput: makeShadcnAutoIdInput({ Input, Label }),
+      textInput: makeShadcnAutoTextInput({ Input, Label }),
+      numberInput: makeShadcnAutoNumberInput({ Input, Label }),
+      encryptedInput: makeShadcnAutoEncryptedStringInput({ Input, Label, Button }),
+      passwordInput: makeShadcnAutoPasswordInput({ Input, Label, Button }),
+      booleanInput: makeShadcnAutoBooleanInput({ Checkbox, Label }),
+    }),
+    [Input, Label, Checkbox, Button]
+  );
```

Comment:
I don't think its legal to use `useMemo` outside a component, and this factory function is outside a component. I think you can just make these local variables though, and then the component below will close over them!